core/src/main/scala/com/microsoft/azure/synapse/ml/logging/Usage/FabricTokenServiceClient.scala

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.logging.Usage
+
+import java.util.UUID
+import java.net.URL
+import java.net.InetAddress
+import java.lang.management.ManagementFactory
+import spray.json.DefaultJsonProtocol.StringJsonFormat
+import com.microsoft.azure.synapse.ml.logging.Usage.FabricConstants._
+import spray.json.{JsArray, JsObject, JsValue, _}
+import com.microsoft.azure.synapse.ml.logging.common.WebUtils.usageGet
+import com.microsoft.azure.synapse.ml.logging.SynapseMLLogging
+
+class FabricTokenServiceClient {
+  val resourceMapping = Map(
+    "https://storage.azure.com" -> "storage",
+    "storage" -> "storage",
+    "https://analysis.windows.net/powerbi/api" -> "pbi",
+    "pbi" -> "pbi",
+    "https://vault.azure.net" -> "keyvault",
+    "keyvault" -> "keyvault",
+    "https://kusto.kusto.windows.net" -> "kusto",
+    "kusto" -> "kusto"
+  )
+
+  private val hostname = InetAddress.getLocalHost.getHostName
+  private val processDetail = ManagementFactory.getRuntimeMXBean.getName
+  private val processName = processDetail.substring(processDetail.indexOf('@') + 1)
```

Comment:
is it guarenteed to be all of the remaining string after the first @?

Reply:
Yes as Scala binary runs on java virtual machine.